### PR TITLE
Feature/146 hazard table dynamic controls

### PR DESCRIPTION
**Pull request by 0qy**

### CODE REVIEW

File: src/service/inversionSolution.service.ts [hunk 1]
```diff
@@ -40,5 +42,34 @@ export const filterData = (
...
+  const rows = data?.node?.rows;
+
+  const forecastTimes = new Set<string>();
+  const bgSeismicity = new Set<string>();
+  const pga = new Set<string>();
+  const gmpe = new Set<string>();
+  const locations = new Set<string>();
+
+  rows?.map((row) => {
+    if (row && row !== null) {
```
Comment (wooneh) [inline src/service/inversionSolution.service.ts:59]: if (row) already tests for nullability so row !== null is a bit redundant

File: src/service/inversionSolution.service.ts [hunk 2]
```diff
@@ -40,5 +42,34 @@ export const filterData = (
...
+      row[0] !== null && forecastTimes.add(row[0] as string);
```
Comment (wooneh) [inline src/service/inversionSolution.service.ts:60]: I doubt the compiler will complain if you removed the cast to string, since there is a null check

**Review by chrisbc — APPROVED** [review]

File: src/service/inversionSolution.service.ts (hunk 1 above)
Comment (0qy) [reply src/service/inversionSolution.service.ts:59]: Thanks that's true changed it now

File: src/service/inversionSolution.service.ts (hunk 2 above)
Comment (0qy) [reply src/service/inversionSolution.service.ts]: Just changed this too thanks :)

**Review by chrisbc — APPROVED** [review]

**Pull request merged**